manager/dispatcher/assignments.go

```diff
@@ -45,39 +34,83 @@ func newAssignmentSet(log *logrus.Entry, dp *drivers.DriverProvider) *assignment
 	}
 }
 
+func assignSecret(a *assignmentSet, readTx store.ReadTx, mapKey typeAndID) {
+	a.tasksUsingDependency[mapKey] = make(map[string]struct{})
+	a.log.WithField("resource.type", "secret")
+	secret, err := a.secret(readTx, mapKey.id)
+	if err != nil {
+		a.log.WithFields(logrus.Fields{
+			"secret.id": mapKey.id,
+			"error":     err,
+		}).Debug("failed to fetch secret")
+		return
+	}
+	if secret == nil {
+		a.log.WithFields(logrus.Fields{
+			"secret.id": mapKey.id,
+		}).Debug("secret not found")
+		return
+	}
+	a.changes[mapKey] = &api.AssignmentChange{
+		Assignment: &api.Assignment{
+			Item: &api.Assignment_Secret{
+				Secret: secret,
+			},
+		},
+		Action: api.AssignmentChange_AssignmentActionUpdate,
+	}
+}
+
+func assignConfig(a *assignmentSet, readTx store.ReadTx, mapKey typeAndID) {
+	a.tasksUsingDependency[mapKey] = make(map[string]struct{})
+	a.log.WithField("resource.type", "config")
```

Comment:
`WithField` returns a logger with that field added to it - think of it as functional programming. You need to use the return value later on in the function, otherwise the call does nothing.